File: contable_uxser_app/lib/domain/entities/sesion_caja.dart
import '../enums/estado_sesion_caja.dart';

class SesionCaja {
  final String id;
  final String empresaId;
  final String usuarioId;
  final DateTime fechaApertura;
  final DateTime? fechaCierre;
  final double valorApertura;
  final double? valorCierreReal;
  final double? valorCierreCalculado;
  final double? diferencia;
  final EstadoSesionCaja estado;

  SesionCaja({
    required this.id,
    required this.empresaId,
    required this.usuarioId,
    required this.fechaApertura,
    this.fechaCierre,
    required this.valorApertura,
    this.valorCierreReal,
    this.valorCierreCalculado,
    this.diferencia,
    required this.estado,
  });

  bool get estaAbierta => estado == EstadoSesionCaja.abierta;
}

File: src/ContableUxser.Domain/Entities/SesionCaja.cs
using ContableUxser.Domain.Common;
using ContableUxser.Domain.Enums;

namespace ContableUxser.Domain.Entities;

public class SesionCaja : BaseEntity
{
    public Guid EmpresaId { get; set; }
    public Guid UsuarioId { get; set; }
    public DateTime FechaApertura { get; set; } = DateTime.UtcNow;
    public DateTime? FechaCierre { get; set; }
    public decimal ValorApertura { get; set; }
    public decimal? ValorCierreReal { get; set; }
    public decimal? ValorCierreCalculado { get; set; }
    public decimal? Diferencia { get; set; }
    public EstadoSesionCaja Estado { get; set; } = EstadoSesionCaja.Abierta;

    public Empresa Empresa { get; set; } = null!;
    public Usuario Usuario { get; set; } = null!;
    public ICollection<Venta> Ventas { get; set; } = new List<Venta>();
}
